src/js/node/zlib.ts

```diff
@@ -86,358 +285,459 @@ ZlibBase.prototype.flush = function (kind, callback) {
+  if (!this.destroyed) {
+    this._level = level;
+    this._strategy = strategy;
+    if (callback) callback();
+  }
 }
-BrotliDecompress.prototype = Object.create(ZlibBase.prototype);
 
-//
+Zlib.prototype.params = function params(level, strategy, callback) {
+  checkRangesOrGetDefault(level, "level", Z_MIN_LEVEL, Z_MAX_LEVEL);
+  checkRangesOrGetDefault(strategy, "strategy", Z_DEFAULT_STRATEGY, Z_FIXED);
+
+  if (this._level !== level || this._strategy !== strategy) {
+    this.flush(Z_SYNC_FLUSH, paramsAfterFlushCallback.bind(this, level, strategy, callback));
+  } else {
+    process.nextTick(callback);
+  }
+};
 
 function Deflate(opts) {
   if (!(this instanceof Deflate)) return new Deflate(opts);
-  Zlib.$call(this, opts, DEFLATE);
+  Zlib.$apply(this, [opts, DEFLATE]);
 }
 Deflate.prototype = Object.create(Zlib.prototype);
 
-//
-
 function Inflate(opts) {
   if (!(this instanceof Inflate)) return new Inflate(opts);
-  Zlib.$call(this, opts, INFLATE);
+  Zlib.$apply(this, [opts, INFLATE]);
 }
 Inflate.prototype = Object.create(Zlib.prototype);
 
-//
+function Gzip(opts) {
+  if (!(this instanceof Gzip)) return new Gzip(opts);
+  Zlib.$apply(this, [opts, GZIP]);
+}
+Gzip.prototype = Object.create(Zlib.prototype);
+
+function Gunzip(opts) {
+  if (!(this instanceof Gunzip)) return new Gunzip(opts);
+  Zlib.$apply(this, [opts, GUNZIP]);
+}
+Gunzip.prototype = Object.create(Zlib.prototype);
 
 function DeflateRaw(opts) {
+  if (opts && opts.windowBits === 8) opts.windowBits = 9;
   if (!(this instanceof DeflateRaw)) return new DeflateRaw(opts);
-  Zlib.$call(this, opts, DEFLATERAW);
+  Zlib.$apply(this, [opts, DEFLATERAW]);
 }
 DeflateRaw.prototype = Object.create(Zlib.prototype);
 
-//
-
 function InflateRaw(opts) {
   if (!(this instanceof InflateRaw)) return new InflateRaw(opts);
-  Zlib.$call(this, opts, INFLATERAW);
+  Zlib.$apply(this, [opts, INFLATERAW]);
 }
 InflateRaw.prototype = Object.create(Zlib.prototype);
 
-//
-
-function Gzip(opts) {
-  if (!(this instanceof Gzip)) return new Gzip(opts);
-  Zlib.$call(this, opts, GZIP);
+function Unzip(opts) {
+  if (!(this instanceof Unzip)) return new Unzip(opts);
+  Zlib.$apply(this, [opts, UNZIP]);
 }
-Gzip.prototype = Object.create(Zlib.prototype);
-
-//
+Unzip.prototype = Object.create(Zlib.prototype);
 
-function Gunzip(opts) {
-  if (!(this instanceof Gunzip)) return new Gunzip(opts);
-  Zlib.$call(this, opts, GUNZIP);
+function createConvenienceMethod(ctor, sync) {
+  if (sync) {
+    return function syncBufferWrapper(buffer, opts) {
+      return zlibBufferSync(new ctor(opts), buffer);
+    };
+  }
+  return function asyncBufferWrapper(buffer, opts, callback) {
+    if (typeof opts === "function") {
+      callback = opts;
+      opts = {};
+    }
+    return zlibBuffer(new ctor(opts), buffer, callback);
+  };
 }
-Gunzip.prototype = Object.create(Zlib.prototype);
 
-//
+const kMaxBrotliParam = MathMax(
+  ...ArrayPrototypeMap.$call(ObjectKeys(constants), key =>
```

Comment:
can you copy paste the constant once instead of doing this every time bun starts?